src/test/java/com/algorand/algosdk/cucumber/shared/TransactionSteps.java

```diff
@@ -120,9 +129,30 @@ public void sign_the_transaction() throws NoSuchAlgorithmException {
         signedTransaction = base.signTransaction(applicationTransaction);
     }
 
+    @Then("fee field is in txn")
+    public void fee_field_is_in_txn() throws IOException {
+        if (signedTransaction == null) return;
+
+        byte[] encodedTxn = Encoder.encodeToMsgPack(signedTransaction);
+        Map<String,Object> sigtxn = Encoder.decodeFromMsgPack(encodedTxn,Map.class);
+        Map<String,Object> txn = (Map<String,Object>) sigtxn.get("txn");
+        assertThat(txn).containsKey("fee");
+    }
+
+    @Then("fee field not in txn")
+    public void  fee_field_not_in_txn() throws IOException {
+        if (signedTransaction == null) return;
+
+        byte[] encodedTxn = Encoder.encodeToMsgPack(signedTransaction);
+        Map<String,Object> sigtxn = Encoder.decodeFromMsgPack(encodedTxn,Map.class);
+        assertThat(sigtxn).isNotNull();
```

Comment:
Why is this check not in the `fee_field_is_in_txn()` method?

Reply:
Added.